services/shop/src/index.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ Copyright 2022 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import express, { NextFunction, Application, Request, Response } from "express"
+import session from "express-session"
+import MemoryStoreFactory from "memorystore"
+
+import { DSP_HOST, EXTERNAL_PORT, PORT, SHOP_DETAIL, SHOP_HOST, SSP_HOST } from "./env.js"
+import { Order, addOrder, displayCategory, fromSize, getItem, getItems, removeOrder, updateOrder } from "./lib/items.js"
+
+const app: Application = express()
+
+declare module "express-session" {
+  interface SessionData {
+    cart: Order[]
+  }
+}
+
+app.set("trust proxy", 1) // required for Set-Cookie with Secure
+
+// Due to express >= 4 changes, we need to pass express-session to the
+// function memorystore exports in order to extend session.Store:
+const MemoryStore = MemoryStoreFactory(session)
+
+const oneDay = 1000 * 60 * 60 * 24
+app.use(
+  session({
+    name: "__session", // https://firebase.google.com/docs/hosting/manage-cache#using_cookies
+    secret: "THIS IS SECRET FOR DEMO", // replace with your secret at build time.
```

Comment:
according to https://expressjs.com/en/resources/middleware/session.html

`Using a secret that cannot be guessed will reduce the ability to hijack a session to only guessing the session ID (as determined by the genid option).`

The impact of hijacking a session of Privacy Sandbox Demos is low (taking over cart, or demo parameters).
no personal informations are stored and linked to the current user session.

long term it would be better to generate a secret at build time and pass it as an environment variable.